### PR TITLE
React to AuthN changes in MVC

### PR DESCRIPTION
**Pull request by HaoK**
Pretty much just renames right now for MVC, with a few changes in the test code that has an Auth Middleware

cc @yishaigalatzer

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/ChallengeResult.cs [hunk 1]
```diff
@@ -36,18 +36,18 @@ public ChallengeResult(string authenticationType, AuthenticationProperties prope
 
         public ChallengeResult(IList<string> authenticationTypes, AuthenticationProperties properties)
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/ChallengeResult.cs]: Let's keep the this consistent with `AuthenticationSchemes` too

File: src/Microsoft.AspNet.Mvc.Core/project.json [hunk 2]
```diff
@@ -12,7 +12,8 @@
     "Microsoft.AspNet.Mvc.Common": { "version": "6.0.0-*", "type": "build" },
     "Microsoft.AspNet.Mvc.ModelBinding": "6.0.0-*",
     "Microsoft.AspNet.Routing": "1.0.0-*",
-    "Microsoft.AspNet.Security": "1.0.0-*",
+    "Microsoft.AspNet.Authentication": "1.0.0-*",
+    "Microsoft.AspNet.Authorization": "1.0.0-*",
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/project.json]: please sort

File: test/WebSites/FiltersWebSite/AuthorizeBasicMiddleware.cs [hunk 3]
```diff
@@ -4,10 +4,11 @@
 using System;
 using System.Security.Claims;
 using System.Threading.Tasks;
+using Microsoft.AspNet.Authentication;
+using Microsoft.AspNet.Authentication.Infrastructure;
```
Comment (rynowak) [inline test/WebSites/FiltersWebSite/AuthorizeBasicMiddleware.cs]: What's pulling this in?

Comment (HaoK) [inline test/WebSites/FiltersWebSite/AuthorizeBasicMiddleware.cs]: The middleware base class pulls that in, Infrastructure is getting nuked soon and rolled into the base namespace which will get rid of one using.

File: src/Microsoft.AspNet.Mvc.Core/project.json (hunk 2 above)
Comment (HaoK) [inline src/Microsoft.AspNet.Mvc.Core/project.json]: Fixed

File: src/Microsoft.AspNet.Mvc.Core/ActionResults/ChallengeResult.cs (hunk 1 above)
Comment (HaoK) [inline src/Microsoft.AspNet.Mvc.Core/ActionResults/ChallengeResult.cs]: Switched to schemes

**Comment by HaoK** [comment]
Merged 46aaf790c4fcb0f7a45e68d76b5cd0a2bf1a9413

**Pull request closed without merging**